java/fury-core/src/main/java/org/apache/fury/resolver/ClassResolver.java

```diff
@@ -728,13 +731,17 @@ private void addSerializer(Class<?> type, Serializer<?> serializer) {
       }
       classInfo = classInfoMap.get(type);
     }
+
     if (classInfo == null || typeTag != null || classId != classInfo.classId) {
-      classInfo = new ClassInfo(this, type, typeTag, serializer, classId);
-    } else {
-      classInfo.serializer = serializer;
+      classInfo = new ClassInfo(this, type, typeTag, null, classId);
+      classInfoMap.put(type, classInfo);
+      if (registered) {
+        registeredId2ClassInfo[classId] = classInfo;
```

Comment:
`registered == true` means that ClassInfo has been re-created, so `registeredId2ClassInfo` needs to be updated.